packages/client/src/client.ts

```diff
@@ -1,48 +1,61 @@
-import { EvaluationContext, OpenFeatureError } from '@openfeature/shared';
-import { SafeLogger } from '@openfeature/shared';
 import {
   ClientMetadata,
   ErrorCode,
+  EvaluationContext,
   EvaluationDetails,
   FlagValue,
   FlagValueType,
   HookContext,
   JsonValue,
   Logger,
+  OpenFeatureError,
+  ProviderStatus,
   ResolutionDetails,
-  StandardResolutionReasons
+  SafeLogger,
+  StandardResolutionReasons,
 } from '@openfeature/shared';
 import { OpenFeature } from './open-feature';
-import { Client, FlagEvaluationOptions, Handler, Hook, OpenFeatureEventEmitter, Provider, ProviderEvents } from './types';
+import {
+  Client,
+  FlagEvaluationOptions,
+  Handler,
+  Hook,
+  OpenFeatureEventEmitter,
+  Provider,
+  ProviderEvents,
+} from './types';
 
 type OpenFeatureClientOptions = {
   name?: string;
   version?: string;
 };
 
 export class OpenFeatureClient implements Client {
-  readonly metadata: ClientMetadata;
   private _hooks: Hook[] = [];
   private _clientLogger?: Logger;
 
   constructor(
     // functions are passed here to make sure that these values are always up to date,
-    // and so we don't have to make these public properties on the API class. 
+    // and so we don't have to make these public properties on the API class.
     private readonly providerAccessor: () => Provider,
-    private readonly providerReady: () => boolean,
     private readonly events: () => OpenFeatureEventEmitter,
     private readonly globalLogger: () => Logger,
-    options: OpenFeatureClientOptions,
-  ) {
-    this.metadata = {
-      name: options.name,
-      version: options.version,
-    } as const;
+    private readonly options: OpenFeatureClientOptions
+  ) {}
+
+  get metadata(): ClientMetadata {
+    return {
+      name: this.options.name,
+      version: this.options.version,
+      provider: this.providerAccessor().metadata,
```

Comment:
FYI @toddbaert, I talked to @lukas-reining about adding this. I don't think it's specced out but it makes sense to include provider metadata in the client now that we have multi-provider support.

Reply:
I see why we'd want this. Though I wonder if it might be better to call it `providerMetadata` to make it clearer.

Reply:
Changed it.